File: TrakStoreApi/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TrakStoreApi.Models
{
    public class Comment
    {   
        [Key]
        public int Id { get; set; }
        public required DateTime Date { get; set; }
        public required string Text { get; set; }
        public required int Star { get; set; } 

        // Çoktan çoğa ilişki -> Bir yorum birden fazla ürünle ilişkilendirilebilir.
        [JsonIgnore]
        public ICollection<CommentProduct> CommentProducts { get; set; }
    }
}

File: lib/Models/comment.dart
class Comment {
  int id;
  DateTime date;
  String text;
  int star;

  Comment({
    required this.id,
    required this.text,
    required this.star,
    required this.date,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['productId'],
      date: DateTime.parse(json['comment']["date"]),
      star: json["comment"]['star'],
      text: json["comment"]['text'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'date': date, 'text': text, 'star': star};
  }
}
